subprojects/plugins/src/integTest/groovy/org/gradle/java/fixtures/AbstractJavaTestFixturesIntegrationTest.groovy

```diff
@@ -0,0 +1,393 @@
+            ":compileJava",
+            ":compileTestFixturesJava",
+            ":compileTestJava",
+            *producedJars
+        )
+        notExecuted(*skippedJars)
+
+        when:
+        succeeds "test"
+
+        then:
+        def expectedJars = [':jar', ':testFixturesJar'] - producedJars
+        executedAndNotSkipped(*expectedJars)
+    }
+
+    def "test fixtures implementation dependencies to not leak into the test compile classpath"() {
+        buildFile << """
+            apply plugin: 'java-test-fixtures'
+        
+            dependencies {
+                testFixturesImplementation 'org.apache.commons:commons-lang3:3.9'
+            }
+        """
+        addPersonDomainClass()
+        addPersonTestFixtureUsingApacheCommons()
+        addPersonTestUsingTestFixtures()
+        file("src/test/java/org/Leaking.java") << """
+            package org;
+            import org.apache.commons.lang3.StringUtils;
+            
+            public class Leaking {
+            }
+        """
+        when:
+        fails 'compileTestJava'
+
+        then:
+        failure.assertHasCause("Compilation failed")
+        errorOutput.contains("package org.apache.commons.lang3 does not exist")
+    }
+
+    def "test fixtures api dependencies are visible on the test compile classpath"() {
+        buildFile << """
+            apply plugin: 'java-test-fixtures'
+        
+            dependencies {
+                testFixturesApi 'org.apache.commons:commons-lang3:3.9'
+            }
+        """
+        addPersonDomainClass()
+        addPersonTestFixtureUsingApacheCommons()
+        addPersonTestUsingTestFixtures()
+        file("src/test/java/org/Leaking.java") << """
+            package org;
+            import org.apache.commons.lang3.StringUtils;
+            
+            public class Leaking {
+            }
+        """
+
+        expect:
+        succeeds 'compileTestJava'
+    }
+
+    def "can consume test fixtures of subproject"() {
+        settingsFile << """
+            include 'sub'
+        """
+        file("sub/build.gradle") << """
+            apply plugin: 'java-test-fixtures'
+        """
+        buildFile << """
+            dependencies {
+                testImplementation(testFixtures(project(":sub")))
+            }           
+        """
+        addPersonDomainClass("sub")
+        addPersonTestFixture("sub")
+        // the test will live in the current project, instead of "sub"
+        // which demonstrates that the test fixtures are exposed
+        addPersonTestUsingTestFixtures()
+
+        when:
+        succeeds ':compileTestJava'
+
+        then:
+        executedAndNotSkipped(
+            ":sub:compileTestFixturesJava"
+        )
+    }
+
+    def "can publish test fixtures"() {
+        FeaturePreviewsFixture.enableGradleMetadata(settingsFile)
+
+        buildFile << """
+            apply plugin: 'maven-publish'
+            apply plugin: 'java-test-fixtures'
```

Comment:
This is not directly related to this feature, but I wonder if a user has all the entry points he needs to do something similar for his own thing. I'll work on setting up a sample proving (or disproving) that.

Reply:
I don't understand what you are talking about :)